using_restricted_array.rb

```diff
@@ -6,6 +6,7 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
+
```

Comment:
You could optimize by just using one variable here: return i and get rid of length, or index using length and get rid of i.